src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyResponseImpl.java

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.proton.transport.proxy.impl;
+
+import com.microsoft.azure.proton.transport.proxy.HttpStatusLine;
+import com.microsoft.azure.proton.transport.proxy.ProxyResponse;
+import com.microsoft.azure.proton.transport.ws.WebSocket.WebSocketFrameReadState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents an HTTP response from a proxy.
+ *
+ * @see <a href="https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html">RFC2616</a>
+ */
+public final class ProxyResponseImpl implements ProxyResponse {
+    private static final String CONTENT_LENGTH = "Content-Length";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyResponseImpl.class);
+
+    private final HttpStatusLine status;
+    private final Map<String, List<String>> headers;
+    private final ByteBuffer contents;
+
+    private ProxyResponseImpl(HttpStatusLine status, Map<String, List<String>> headers, ByteBuffer contents) {
+        this.status = status;
+        this.headers = headers;
+        this.contents = contents;
+    }
+
+    static ProxyResponse create(ByteBuffer buffer) {
```

Comment:
I was curious why we opted to use this static factory method rather than a constructor (ie. new ProxyResponseImpl(ByteBuffer)

I was expecting that an improperly formed ProxyResponse would throw an exception, in a few cases for this factory method, I see some instances of returning null.

Reply:
When I tried to use new ProxyResponseImpl(ByteBuffer), I found it is not very clear to assign values for its private variables(etc. status,  headers, contents) inside the whole parsing logic. So we can have a the private constuctor ` ProxyResponseImpl(status, headers, content) to assign all values at once. Because Java disallow us if we don't call another constuctor in the first line, we could use a static factory as one way. It is also make sense use factory to create object when create logic is complex.

And for returning null, I changed it to throw an Exception instead. Thanks for comments!